### PR TITLE
Fix and test MBEDTLS_PSA_INJECT_ENTROPY

### LINKED ISSUE #7516: MBEDTLS_PSA_INJECT_ENTROPY breaks the build since 2.26
The build option `MBEDTLS_PSA_INJECT_ENTROPY` has been broken since Mbed TLS 2.26 (and still is as of 2.28.3 and 3.4.0).

To reproduce:
```
scripts/config.py set MBEDTLS_PSA_INJECT_ENTROPY
scripts/config.py set MBEDTLS_ENTROPY_NV_SEED
scripts/config.py set MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES
make lib
```

Expected: success. (Building a program would cause a link error due to missing platform functions, that part is expected.)

Actual:
```
…
library/psa_crypto.c: In function ‘mbedtls_psa_random_init’:
library/psa_crypto.c:6051:33: error: ‘mbedtls_nv_seed_poll’ undeclared (first use in this function); did you mean ‘mbedtls_nv_seed_read’?
 6051 |                                 mbedtls_nv_seed_poll, NULL,
      |                                 ^~~~~~~~~~~~~~~~~~~~
      |                                 mbedtls_nv_seed_read
```

This option is tricky to use and not well documented, but it is used in Mbed OS (where it was integrated by people who were on the Mbed Crypto team). We want to replace it by a properly integrated mechanism under the PSA API (which may be backward compatible — TBD), but that mechanism isn't implemented yet. In the meantime, `MBEDTLS_PSA_INJECT_ENTROPY` as currently implemented needs to work.

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
* Fix #7516: fix the build when `MBEDTLS_PSA_INJECT_ENTROPY` is enabled.
* Add a test component with `MBEDTLS_PSA_INJECT_ENTROPY` enabled. This required instrumenting the test framework to perform the necessary entropy injection.
* We had tests for `MBEDTLS_PSA_INJECT_ENTROPY`, but they'd never been enabled in any CI build. Make sure they pass and don't leave the system in a state that breaks subsequent tests.
* Add some checks in the tests that would break if we made an incompatible change to the seed file. Fixes https://github.com/Mbed-TLS/mbedtls/issues/7517.

## Gatekeeper checklist

- [x] **changelog** provided
- [ ] **backport** in #7965
- [x] **tests** provided

### CODE REVIEW

File: tests/configs/user-config-for-test.h [hunk 1]
```diff
@@ -55,3 +55,23 @@
 #define MBEDTLS_PSA_ACCEL_ALG_HMAC
 
 #endif  /* PSA_CRYPTO_DRIVER_TEST_ALL */
+
+
+
```
Comment (tom-cosgrove-arm) [inline tests/configs/user-config-for-test.h:60]: (very minor) any reason why quite so many blank lines?

**Review by tom-cosgrove-arm — DISMISSED** [review]
LGTM

File: tests/configs/user-config-for-test.h [hunk 2]
```diff
@@ -55,3 +55,23 @@
...
+ * (These could have been provided as library functions, but for historical
+ * reasons, they weren't, and so each integrator has to provide a copy
+ * of these functions.)
+ *
+ * Provide implementations of these functions for testing. */
+#include <stddef.h>
+int mbedtls_test_inject_entropy_seed_read(unsigned char *buf, size_t len);
+int mbedtls_test_inject_entropy_seed_write(unsigned char *buf, size_t len);
+#define MBEDTLS_PLATFORM_NV_SEED_READ_MACRO mbedtls_test_inject_entropy_seed_read
+#define MBEDTLS_PLATFORM_NV_SEED_WRITE_MACRO mbedtls_test_inject_entropy_seed_write
```
Comment (ronald-cron-arm) [inline tests/configs/user-config-for-test.h:76]: For my information, in which circumstances, mbedtls_test_inject_entropy_seed_write is called (I looked at that but did not really figure out)?

**Review by ronald-cron-arm — CHANGES_REQUESTED** [review]
It  looks almost good to me. Just some comments that do not seem to be right to me and one question.

File: tests/configs/user-config-for-test.h (hunk 2 above)
Comment (gilles-peskine-arm) [reply tests/configs/user-config-for-test.h:76]: In `include/mbedtls/platform.h`, in the configurations supported for this test (`MBEDTLS_ENTROPY_NV_SEED` set and `MBEDTLS_PLATFORM_NV_SEED_ALT` unset):
```
#define mbedtls_nv_seed_read    MBEDTLS_PLATFORM_NV_SEED_READ_MACRO
#define mbedtls_nv_seed_write   MBEDTLS_PLATFORM_NV_SEED_WRITE_MACRO
```

`mbedtls_nv_seed_write` is called in `mbedtls_entropy_update_nv_seed` in `entropy.c`, which in turn is called from `mbedtls_entropy_func`, which is the function to use for seeding a DRBG instance from entropy.

**Comment by gilles-peskine-arm** [comment]
The 2.28 backport is up: https://github.com/Mbed-TLS/mbedtls/pull/7965

File: tests/configs/user-config-for-test.h (hunk 2 above)
Comment (ronald-cron-arm) [reply tests/configs/user-config-for-test.h:76]: okay, I see, thanks.

**Review by ronald-cron-arm — APPROVED** [review]
LGTM

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Pull request merged**